frontend/src/lib/utils.tsx

```diff
@@ -2042,3 +2042,36 @@ export const getJSHeapMemory = (): {
     }
     return {}
 }
+
+interface PreviewFlagsV2Config {
+    rolloutPercentage: number
+    includedHashes?: Set<string>
+    excludedHashes?: Set<string>
+}
+
+function hashString(str: string): string {
```

Comment:
Just out of curiosity, why did you hand roll a hash function? The purpose was to hide the api keys. I’m not familiar enough with your implementation to understand if it’s one-way secure.

Reply:
I wanted to do something simple; but you're right, I should pick a tried-and-true implementation. I've updated it.

Reply:
Ha! I'm allergic to hand-rolled crypto. 😛